tests/e2e/fixtures/metrics/custom-queries-with-predicate-query.yaml

```diff
@@ -0,0 +1,89 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: monitoring-01
+  labels:
+    e2e: metrics
+data:
+  queries.yaml: |
+    pg_predicate_query_return_true:
+      query: "SELECT 42 as fixed"
+      predicate_query: "SELECT true as result"
+      primary: false
+      metrics:
+        - fixed:
+            usage: "GAUGE"
+            description: "Always 42, used to test predicate_query"
+    pg_predicate_query_return_false:
+      query: "SELECT 42 as fixed"
+      predicate_query: "SELECT false as result"
+      primary: false
+      metrics:
+        - fixed:
+            usage: "GAUGE"
+            description: "Always 42, used to test predicate_query"
+    pg_predicate_query_return_null_as_false:
+      query: "SELECT 42 as fixed"
+      predicate_query: "SELECT null as result"
+      primary: false
+      metrics:
+        - fixed:
+            usage: "GAUGE"
+            description: "Always 42, used to test predicate_query"        
+    pg_predicate_query_return_no_rows:
+      query: "SELECT 42 as fixed"
+      predicate_query: "SELECT true as result WHERE 1 <> 1"
+      primary: false
+      metrics:
+        - fixed:
+            usage: "GAUGE"
+            description: "Always 42, used to test predicate_query"
+    pg_predicate_query_return_true_and_multiple_columns:
+      query: "SELECT 42 as fixed"
+      predicate_query: "SELECT true as result, 0 as column_one, 'something' as column_two"
+      primary: false
+      metrics:
+        - fixed:
+            usage: "GAUGE"
+            description: "Always 42, used to test predicate_query"
+    pg_predicate_query_return_false_and_multiple_columns:
+      query: "SELECT 42 as fixed"
+      predicate_query: "SELECT false as result, 0 as column_one, 'something' as column_two"
+      primary: false
+      metrics:
+        - fixed:
+            usage: "GAUGE"
+            description: "Always 42, used to test predicate_query"
+    pg_predicate_query_return_true_and_multiple_rows:
+      query: "SELECT 42 as fixed"
+      predicate_query: "SELECT result FROM (SELECT true as result UNION SELECT false as result) order by result desc"
```

Comment:
This fails with ` ERROR: subquery in FROM must have an alias (SQLSTATE 42601)`

It could be formulated as

```
SELECT result FROM (VALUES (true), (false)) AS r(result) ORDER BY 1  DESC;
```

which produces

```
14/main craig@craig > SELECT result FROM (VALUES (true), (false)) AS r(result) ORDER BY 1 DESC;
 result 
--------
 t
 f
(2 rows)
```

Reply:
Thanks a lot for your suggestion, I will update the PR according!

Reply:
Solved!